zipline/data/us_equity_pricing.py

```diff
@@ -355,15 +344,30 @@ def iterator(iterator=iterator, assets=set(assets)):
             last_row[asset_key] = total_rows + nrows - 1
             total_rows += nrows
 
+            table_day_to_session = compose(
+                self._calendar.minute_to_session_label,
+                partial(Timestamp, unit='s', tz='UTC'),
+            )
+            asset_first_day = table_day_to_session(table['day'][0])
+            asset_last_day = table_day_to_session(table['day'][-1])
+
+            asset_sessions = sessions[
+                sessions.slice_indexer(asset_first_day, asset_last_day)
+            ]
+            assert len(table) == len(asset_sessions), (
+                'Got {} rows for daily bars table with first day={}, last '
+                'day={}, expected {} rows.'.format(
+                    len(table),
+                    asset_first_day.date(),
+                    asset_last_day.date(),
```

Comment:
Wouldn't block on it. But we could put the different values between `table['days']` and `sessions` in to the error message.